src/components/custom/types.ts

```diff
@@ -40,18 +42,31 @@ export interface ICustomElementJsonSchema<T> {
 /**
  * base schema for custom fields
  */
-export interface ICustomFieldJsonSchema<T, V = undefined, U = undefined> extends ICustomElementJsonSchema<T> {
+export interface IBaseCustomFieldSchema<T, V = undefined, U = undefined> extends ICustomElementJsonSchema<T> {
 	validation?: (V | U | IYupValidationRule)[];
 	/** render conditions
 	 * - need to fulfil at least 1 object in array (OR condition)
 	 * - in order for an object to be valid, need to fulfil all conditions in that object (AND condition) */
 	showIf?: TRenderRules[] | undefined;
 }
 
+// =============================================================================
+// CUSTOM ELEMENT / FIELD PROPS
+// =============================================================================
 /**
  * common props for all custom elements / fields
  */
-export interface IGenericCustomFieldProps<T = TFrontendEngineFieldSchema> {
+export interface IGenericCustomElementProps<T = TCustomComponentSchema> {
+	id: string;
+	schema: T;
+}
+
+/**
+ * common props for all custom fields
+ */
+export interface IGenericCustomFieldProps<T = IFilterCheckboxSchema>
```

Comment:
can `T` be left as a required generic parameter? or is the default required?

Reply:
Ok I have removed it and did so for the other generic props as well because I realised these generic props are only used by the individual fields / elements / custom components.